core/kt-osrd-utils/src/main/kotlin/fr/sncf/osrd/utils/DistanceRangeMapImpl.kt

```diff
@@ -4,20 +4,83 @@ import com.google.common.collect.RangeMap
 import fr.sncf.osrd.utils.units.Distance
 import fr.sncf.osrd.utils.units.MutableDistanceArrayList
 import java.util.function.BiFunction
+import java.util.PriorityQueue
 import kotlin.math.min
 
 data class DistanceRangeMapImpl<T>(
     private val bounds: MutableDistanceArrayList,
     private val values: MutableList<T?>,
 ) : DistanceRangeMap<T> {
 
-    constructor() : this(MutableDistanceArrayList(), ArrayList())
+    constructor(entries: List<DistanceRangeMap.RangeMapEntry<T>> = emptyList())
+      : this(MutableDistanceArrayList(), ArrayList()) {
+            putMany(entries)
+    }
 
     /** Sets the value between the lower and upper distances */
     override fun put(lower: Distance, upper: Distance, value: T) {
         putOptional(lower, upper, value)
     }
 
+    /**
+    Sets many values more efficiently than many calls to `put`
+
+    The idea here is to build the map from scratch, by iterating over sorted bounds and maintaining entries priority.
+
+    Another idea would be to use a temporary (so we can free the memory later) tree-like structure (RangeMaps lib).
+    */
+    override fun putMany(entries: List<DistanceRangeMap.RangeMapEntry<T>>) {
+        // Order matters and existing entries should come first.
+        val allEntries = asList() + entries
+
+        // Start from scratch.
+        values.clear()
+        bounds.clear()
+
+        // Build a sorted list of bounds, while keeping track of entries order.
+        val boundEntries = mutableListOf<Pair<Distance, Int>>()
+        for ((index, entry) in allEntries.withIndex()) {
+            boundEntries.add(Pair(entry.lower, index))
+            boundEntries.add(Pair(entry.upper, index))
+        }
+        boundEntries.sortWith(
+            Comparator<Pair<Distance, Int>>{
+                a, b -> a.first.compareTo(b.first)
+            }.thenBy{it.second}
+        )
+
+        // Relevant entries for the interval we're building. Early entries have low priority.
+        val entryQueue = PriorityQueue<Int>{i, j -> j-i}
+        // Value over the interval we're building.
+        var value: T? = null
+        // Start of the interval we're building.
+        var start: Distance? = null
+        for((bound, index) in boundEntries) {
```

Comment:
```suggestion
        for ((bound, index) in boundEntries) {
```

The code style isn't entirely consistent across the file. If you use intellij, you can press ctrl+shift+alt+L to reformat the file.

We don't (yet) have a style enforced by the CI, but most of the files are formatted with the default intellij formatting.